packages/ng/tooltip/trigger/tooltip-trigger.directive.ts

```diff
@@ -1,156 +1,339 @@
+			positionStrategy: this.legacyPositionBuilder(),
+			scrollStrategy: this.#overlay.scrollStrategies.close(),
+			disposeOnNavigation: true,
+		});
+		const portal = new ComponentPortal(LuTooltipPanelComponent);
+		const ref = this.overlayRef.attach(portal);
+		if (this.luTooltip) {
+			ref.instance.content = this.luTooltip;
+		} else if (this.luTooltipWhenEllipsis) {
+			ref.instance.content = this.#host.nativeElement.innerText;
 		}
+		ref.instance.id = this.ariaDescribedBy;
+		// On tooltip leave => trigger close
+		ref.instance.mouseLeave$.pipe(takeUntilDestroyed(ref.instance.destroyRef)).subscribe(() => this.close$.next());
+		// On tooltip enter => trigger open to keep it opened
+		ref.instance.mouseEnter$.pipe(takeUntilDestroyed(ref.instance.destroyRef)).subscribe(() => this.open$.next());
 	}
 
-	ngAfterViewInit() {
-		this._checkTarget();
-	}
-	ngOnDestroy() {
-		this._cleanUpSubscriptions();
-		if (this._popoverOpen) {
-			this.closePopover();
+	private closeTooltip(): void {
+		if (this.overlayRef) {
+			this.overlayRef.detach();
+			delete this.overlayRef;
 		}
-		this.destroyPopover();
-	}
-	protected _emitOpen(): void {
-		this.onOpen.emit();
-	}
-	protected _emitClose(): void {
-		this.onClose.emit();
 	}
 
-	protected override _createOverlay(): OverlayRef {
-		if (!this._overlayRef) {
-			this._portal = new ComponentPortal(LuTooltipPanelComponent, this._viewContainerRef);
-			const config = this._getOverlayConfig();
-			this._subscribeToPositions(config.positionStrategy as FlexibleConnectedPositionStrategy);
-			this._overlayRef = this._overlay.create(config);
+	private setAccessibilityProperties(tabindex: number | null): void {
+		if (this.#disabled || (this.luTooltipWhenEllipsis && !this.hasEllipsis())) {
+			this.#renderer.removeAttribute(this.#host.nativeElement, 'tabindex');
+			return;
 		}
 
-		return this._overlayRef;
+		const tag = this.#host.nativeElement.tagName.toLowerCase();
+		const nativelyFocusableTags = ['a', 'button', 'input', 'select', 'textarea'];
+		const isNatevelyFocusableTag = nativelyFocusableTags.includes(tag);
+
+		const hasATabIndex = this.#host.nativeElement.getAttribute('tabindex') !== null;
+
+		if (!isNatevelyFocusableTag && !hasATabIndex) {
+			this.#renderer.setAttribute(this.#host.nativeElement, 'tabindex', tabindex.toString());
+		}
 	}
 
-	protected override _attachPortalToOverlay(): void {
-		const componentRef = this._overlayRef.attach(this._portal);
-		this._panel = componentRef.instance;
-		this._panel.content = this._tooltipContent;
+	/**
+	 * Hacky af but let's explain everything
+	 * This method checks for ellipsis by cloning the node and checking its width against original element.
+	 *
+	 * We used to do this using scrollWidth but the thing is, it's a rounded value. Sometimes,
+	 * you'd get true while it should be false and vice-versa, because of rounding.
+	 *
+	 * We could also use getBoundingClientRect() but it only considers text content, meaning that if ellipsis is caused by
```

Comment:
Paddings are included in `getBoundingClientRect` as well as borders. I don't get why we should clone the element. `boundingRect` on host element should be enough?

Reply:
There's a more detailed answer than I could provide here: https://stackoverflow.com/a/64747288/4102561